### DISCUSSION TITLE
Adopting redis version 4

### DISCUSSION THREAD

**Discussion by omer-aydin95** [animir/node-rate-limiter-flexible#167] [Q&A] [answered]
Hi,

First of all, thanks for the library. We use and love it. However, we want to upgrade our `redis` library to version `4.1.0` and we wonder that when the `redis v4` would be supported without using `lagacyMode: true`. Is it in your plans soon?

Kind regards.

**Comment by animir** [comment, marked as answer]
Hey @19XLR95. It is good to know you find this package useful, thank you!

As far as I know, nobody started working on RateLimiterRedis upgrade yet. It is definitely should be done and it is in [v3 notes](https://github.com/animir/node-rate-limiter-flexible/issues/141). Unfortunately, I don't have free time capacity for that now and most likely I won't have it in the next two months.

If somebody starts refactoring for `redis` package v4.x, please, let the community know.

**Reply by omer-aydin95** [reply]
Thanks for the response. I will keep my eye on this.

**Reply by rahil-p** [reply]
@19XLR95 @animir 

I've opened a PR (#176) that implements a separate limiter for v4 clients.  

I'm not sure if this is the preferred direction, since it would probably better to have the existing limiter support v4 clients.  That said, redis v4 is so different from all the other redis clients that the separation may be warranted.   I'd be happy to hear feedback.

**Reply by animir** [reply]
@rahil-p Thanks, I posted the feedback to your PR.

**Reply by omer-aydin95** [reply]
Thanks for your work
